pkg/controller/jobframework/validation.go

```diff
@@ -71,3 +73,14 @@ func ValidateUpdateForParentWorkload(oldJob, newJob GenericJob) field.ErrorList
 	}
 	return allErrs
 }
+
+func ValidateUpdateForOriginalNodeSelectors(oldJob, newJob GenericJob) field.ErrorList {
+	var allErrs field.ErrorList
```

Comment:
Also validate that the selectors are valid

Reply:
The added value is very limited this being the same validation that makes sure that the annotation can not be set while the job is not suspended, the cost being a reflection based u marshaling.

Reply:
you can just validate that the annotation correctly unmarshals.